spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/procedures/RemoveOrphanFilesProcedure.java

```diff
@@ -120,6 +147,13 @@ public InternalRow[] call(InternalRow args) {
       if (fileListView != null) {
         action.compareToFileList(spark().table(fileListView));
       }
+      Preconditions.checkArgument(prefixMismatchMode == null ||
```

Comment:
I feel like this can be simplified a bit.

```
PrefixMismatchMode prefixMismatchMode = args.isNullAt(8) ? null : PrefixMismatchMode.fromString(args.getString(8));
```

Then we can call it like this:

```
if (prefixMismatchMode != null) {
  action.prefixMismatchMode(prefixMismatchMode);
}
```

You will need to add a method to `PrefixMismatchMode` enum.

```
public static PrefixMismatchMode fromString(String modeAsString) {
  Preconditions.checkArgument(modeAsString != null, "Mode should not be null");
  return PrefixMismatchMode.valueOf(modeAsString.toUpperCase(Locale.ENGLISH));
}
```

Reply:
I havent added `fromString`
Instead
```
DeleteOrphanFiles.PrefixMismatchMode prefixMismatchMode = args.isNullAt(8) ? null :
        DeleteOrphanFiles.PrefixMismatchMode.valueOf(args.getString(8)); // IllegalAgumentException for invalid value
...
...
action.prefixMismatchMode(prefixMismatchMode);
```
Is this ok?

Reply:
This will force the user to match the upper case expected by `valueOf`. The solution above is a bit more flexible. We do that in many other enums.